state/governance.js

```diff
@@ -278,6 +297,42 @@ function useGovernance() {
     return proposalId;
   };
 
+  /**
+   * Queues proposals in the Successful state for execution
+   * after timelock delay (currently 2 days)
+   * @param proposalId
+   */
+  const queueProposal = async (proposalId) => {
+    console.assert(proposalId, "proposalId should not be null");
+    const queueABI = find(GovernorAlphaABI, { name: "queue" });
+    const method = governanceContract.method(queueABI);
+    const clause = method.asClause(proposalId);
+    const txResponse = await provider.vendor.sign('tx', [clause])
+                              .signer(address) // This modifier really necessary?
+                              .comment("Sign to queue proposal " + proposalId)
+                              .request();
+
+    const txVisitor = provider.thor.transaction(txResponse.txid);
+    let txReceipt = null;
+    const ticker = provider.thor.ticker();
+
+    // Wait for tx to be confirmed and mined
+    while(!txReceipt) {
+      await ticker.next();
+      txReceipt = await txVisitor.getReceipt();
+      console.log("txReceipt:", txReceipt);
+    }
+
+    // Handle failed tx
+    if (txReceipt.reverted) {
```

Comment:
is there a way to simulate this fail? and handle it?

Reply:
for queuing it's harder to fail. But right now I don't have a reliable way to simulate the failure.

For handling failure conditions (for all smart contract calls) there's a card (task) for it here 
https://trello.com/c/bJtC361w/23-additional-ui-handling-when-proposal-creation-transaction-fails